tasks/drupal.xml

```diff
@@ -188,7 +190,21 @@ Or, you can specify the export file directly:
                 <echo msg="Drupal database connection available." />
             </try>
             <catch>
-                <fail msg="Drupal database connection not available." />
+                <trycatch>
+                    <try>
+                        <!-- @todo This is potentially really sketchy; if you have root access and called your multisite
+                             'mysql' or 'information_schema', would it destroy your MySQL install?
+
+                             This helps with the multisite install process, though... and it will still fail if
+                             permissions are set up properly for mysql...
+                             -->
+                        <drush command="sql-create" assume="yes" />
```

Comment:
Something weird is happening when running `the-build-installer` now.  This doesn't seem to fail on the host when there is no database server running.

```
example > drupal-has-database-connection:

    [drush] Executing '/Users/byronduvall/example/vendor/bin/drush --nocolor --root="/Users/byronduvall/example/web" --uri="https://example.local" --config="/Users/byronduvall/example/drush/drushrc.php" sql-query "SELECT 1"'...
Query failed.                                                            [error]
    [drush] Executing '/Users/byronduvall/example/vendor/bin/drush --nocolor --root="/Users/byronduvall/example/web" --uri="https://example.local" --config="/Users/byronduvall/example/drush/drushrc.php" --yes sql-create'...
    [drush] Creating database drupal. Any possible existing database will be dropped!
    [drush] Do you really want to continue? (y/n): y
     [echo] Drupal database connection available.
Install Drupal now (y,n)?
```

Reply:
This seems to be a bug in Drush.  The result is `false`, but `drush_set_error()` is never called.  This also happens in `drush sql-drop`, and apparently used to happen on `drush sql-query`:

https://github.com/drush-ops/drush/issues/34#issuecomment-326548684

Reply:
I created a PR against Drush 8.x here: https://github.com/drush-ops/drush/pull/3732

Reply:
OK, that one was too aggressive.  See https://github.com/drush-ops/drush/pull/3733

Reply:
A fix for this has been rolled into the 8.x branch!  https://github.com/drush-ops/drush/commit/97f7d2a03ba72369b7e25c96671b1051c7691f2d 🎉